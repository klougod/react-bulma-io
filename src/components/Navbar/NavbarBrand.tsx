import React, { Children, ReactElement, CSSProperties, MouseEvent } from 'react'
import PropTypes from 'prop-types'

export const NavbarBrand = ({ className, menuRef, children }: any) => {
  const handleBurgerClick = (e: MouseEvent<HTMLButtonElement>) => {
    const target = e.target as HTMLTextAreaElement
    if (target.classList.contains('is-active')) {
      target.classList.remove('is-active')
      menuRef.current.classList.remove('is-active')
    } else {
      target.classList.add('is-active')
      menuRef.current.classList.add('is-active')
    }
  }

  const burgerStyle: CSSProperties = {
    backgroundColor: 'transparent',
    border: 0
  }

  const noPointerEvents: CSSProperties = {
    pointerEvents: 'none'
  }

  const NavBurger = () => (
    <button
      onClick={handleBurgerClick}
      className='navbar-burger'
      style={burgerStyle}
      aria-label='abrir menu'
      aria-expanded='false'
    >
      <span aria-hidden='true' style={noPointerEvents} />
      <span aria-hidden='true' style={noPointerEvents} />
      <span aria-hidden='true' style={noPointerEvents} />
    </button>
  )

  return (
    <div className={`navbar-brand ${className}`}>
      {Children.toArray(children).map((item: ReactElement, index) => {
        return React.cloneElement(item, {
          key: index,
          className: `${item.props.className} navbar-item`
        })
      })}
      <NavBurger />
    </div>
  )
}

NavbarBrand.defaultProps = {
  className: '',
  menuRef: null
}

NavbarBrand.propTypes = {
  className: PropTypes.string,
  menuRef: PropTypes.any,
  children: PropTypes.node
}

export default NavbarBrand
