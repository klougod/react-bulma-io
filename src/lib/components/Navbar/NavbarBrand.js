import React, { Children } from 'react'
import PropTypes from 'prop-types'

export const NavbarBrand = ({ className, menuRef, children }) => {
  const handleBurgerClick = e => {
    if (e.target.classList.contains("is-active")) {
      e.target.classList.remove("is-active");
      menuRef.current.classList.remove("is-active");
    } else {
      e.target.classList.add("is-active");
      menuRef.current.classList.add("is-active");
    }
  }

  const burgerStyle = {
    backgroundColor: 'transparent',
    border: 0
  }

  const noPointerEvents = {
    pointerEvents: 'none'
  }

  const NavBurger = () => (
    <button onClick={handleBurgerClick} className="navbar-burger" style={burgerStyle} aria-label="abrir menu" aria-expanded="false">
      <span aria-hidden="true" style={noPointerEvents} />
      <span aria-hidden="true" style={noPointerEvents} />
      <span aria-hidden="true" style={noPointerEvents} />
    </button>
  )

  return (
    <div className={`navbar-brand ${className}`}>
      {Children.toArray(children).map((item, index) => {
        return React.cloneElement(item, {
          key: index,
          className: `${item.props.className} navbar-item`
        })
      })}
      <NavBurger />
    </div>
  )
}

NavbarBrand.defaultProps = {
  className: '',
  menuRef: null
}

NavbarBrand.propTypes = {
  className: PropTypes.string,
  menuRef: PropTypes.any,
  children: PropTypes.node.isRequired
}

export default NavbarBrand