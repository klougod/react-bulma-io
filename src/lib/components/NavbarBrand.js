import React from 'react'
import PropTypes from 'prop-types'

export const NavbarBrand = props => {

  const {
    className,
    children
  } = props

  const handleBurgerClick = e => {
    if (e.target.classList.contains("is-active")) {
      e.target.classList.remove("is-active");
      menuRef.current.classList.remove("is-active");
    } else {
      e.target.classList.add("is-active");
      menuRef.current.classList.add("is-active");
    }
  }


  const navBurger = () => (
    <button onClick={handleBurgerClick} className="navbar-burger" aria-label="abrir menu" aria-expanded="false">
        <span aria-hidden="true" className="has-no-pointer-events" />
        <span aria-hidden="true" className="has-no-pointer-events" />
        <span aria-hidden="true" className="has-no-pointer-events" />
      </button>
  )

  return (
    <div className={`navbar-brand ${className}`}>
      { children.map((item, index) => {
        return React.cloneElement(item, {
          key: index,
          className: `${item.props.className} navbar-item`
        })
      })}
      
    </div>
  )
}

NavbarBrand.defaultProps = {
  className: ''
}

NavbarBrand.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default NavbarBrand