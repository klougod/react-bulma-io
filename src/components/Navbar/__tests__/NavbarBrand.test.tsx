import React from 'react'
import Enzyme from 'enzyme'

import NavbarBrand from '../NavbarBrand'

describe('NavbarEnd component test', function () {
  it('should render without crash without props', function () {
    const navbarEndMock = Enzyme.shallow(<NavbarBrand />)
    expect(navbarEndMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const navbarBrandMockProps = {
      className: 'navbarBrand CLass',
      menuRef: 'navbarBrand'
    }
    const navbarBrandMock = Enzyme.shallow(
      <NavbarBrand {...navbarBrandMockProps} />
    )
    expect(navbarBrandMock).toMatchSnapshot()
  })

  it('should render without crash with children', function () {
    const navbarEndMock = Enzyme.shallow(
      <NavbarBrand>
        <h1>NavbarBrand</h1>
      </NavbarBrand>
    )
    expect(navbarEndMock).toMatchSnapshot()
  })
})
