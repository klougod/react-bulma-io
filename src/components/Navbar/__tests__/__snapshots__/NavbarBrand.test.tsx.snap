// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarEnd component test should render without crash without props 1`] = `
<div
  className="navbar-end "
/>
`;
